### PR TITLE
make std.os.getenvW case insensitive

### LINKED ISSUE #4603: std.process.getEnvVarOwned is case sensitive on Windows
On Windows, environment variables should be case insensitive but `std.process.getEnvVarOwned` returns only case sensitive matches. For example, on my Windows 10 the path variable is "Path" and not "PATH" so `std.ChildProcess.exec` doesn't work when the argument is an executable in the path.

For me this prints "error: FileNotFound" :
```
const std = @import("std");

pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    const args = [_][]const u8{
        "ipconfig",
    };
    _ = try std.ChildProcess.exec(&arena.allocator, args[0..], null, null, std.math.maxInt(usize));
}
```

### PR DESCRIPTION
**Pull request by ilmaria**
Fixes partially #4603. Fixing std.process.getEnvMap is NOT included in this commit.

### CODE REVIEW

**Comment by FireFox317** [comment]
We already have `std.ascii.eqlIgnoreCase` afaik

**Comment by FireFox317** [comment]
Please squash commits into 1 commit using `git rebase -i HEAD~2` :)

**Comment by ilmaria** [comment]
Sure

File: lib/std/os.zig [hunk 1]
```diff
@@ -1185,7 +1185,7 @@ pub fn getenvW(key: [*:0]const u16) ?[:0]const u16 {
         while (ptr[i] != 0) : (i += 1) {}
         const this_value = ptr[value_start..i :0];
 
-        if (mem.eql(u16, key_slice, this_key)) return this_value;
+        if (std.ascii.eqlIgnoreCase(mem.sliceAsBytes(key_slice), mem.sliceAsBytes(this_key))) return this_value;
```
Comment (fengb) [inline lib/std/os.zig]: I'm pretty sure this will erroneously compare the high byte(s) in actual unicode as ascii.

Comment (FireFox317) [reply lib/std/os.zig]: Yup, good catch. Not sure what the best solution would be here :(

Comment (ilmaria) [reply lib/std/os.zig]: Yeah, I think so too. We could either add an ascii check to `std.ascii.eqlIgnoreCase` or a new private function here that makes the check.

**Comment by andrewrk** [comment]
Thanks!

**Pull request closed without merging**